haystack/nodes/preprocessor/preprocessor.py

```diff
@@ -444,3 +457,49 @@ def _find_longest_common_ngram(
             # no common sequence found
             longest = ""
         return longest if longest.strip() else None
+
+    def _sentence_tokenizer(
+        self, text: str, language: str, tokenizer_model_folder: Optional[Union[str, Path]] = None
+    ) -> List[str]:
+        """
+        Tokenize text into sentences.
+        :param text: str, text to tokenize
+        :param language: str, language of text
+        :param tokenizer_model_path: Path, path to tokenizer model
+        :return: list[str], list of sentences
+        """
+        sentences = []
+        language_name = iso639_to_nltk.get(language)
+        tokenizer_model_path = (
+            Path(tokenizer_model_folder) / f"{language}.pickle" if tokenizer_model_folder is not None else None
+        )
+        if tokenizer_model_path is not None:
+            try:
+                sentence_tokenizer = nltk.data.load(str(tokenizer_model_path), format="pickle")
+                sentences = sentence_tokenizer.tokenize(text)
+            except LookupError:
+                logger.error(f"PreProcessor couldn't load sentence tokenizer from {str(tokenizer_model_path)}")
+            except (UnpicklingError, ValueError):
+                logger.error(
+                    f"PreProcessor couldn't determine model format of sentence tokenizer at {str(tokenizer_model_path)}."
+                )
+            if len(sentences) == 0 and language_name is None:
```

Comment:
Just a minor thing here: `language_name is not None` will be `True` if `language_name == ""`. With strings I think it's safer to check with `not language_name`, which will evaluate to `False` if `language_name == ""`. I know that in the current code this can't happen, but I find it a bit more future-proof.

Similarly, you can check for empty lists with `not sentences`.

```suggestion
            if not sentences and not language_name:
```

Reply:
Old habits from C#, checking for null or empty :smiling_face_with_tear: